HEN_HOUSE/egs++/egs_transformations.h

```diff
@@ -215,11 +215,71 @@ class EGS_EXPORT EGS_RotationMatrix {
              (rzx == m.rxz) && (rzy == m.rzy) && (rzz == m.rzz)) ? true : false;
     };
 
-    /*! \brief Returns \c true, if this object is a unity matrix, \a false
+    /*! \brief Returns \c true, if this object is approximately the unit matrix, \a false
       otherwise */
     bool isI() const {
-        return (rxx == 1 && rxy == 0 && rxz == 0 && ryx == 0 && ryy == 1 &&
-                ryz == 0 && rzx == 0 && rzy == 0 && rzz == 1);
+
+        EGS_Float d2;
+        EGS_Float eps2 = epsilon*epsilon;
+
+        // compute square distance to unit matrix, component-by-component;
+        // return as quickly as possible
+
+        // rxy
+        d2 = rxy*rxy;
```

Comment:
Small nitpick. It would be more straight forward to compare e.g. `abs(rxy)` to `eps` instead of squaring both.

Reply:
Which one is more efficient? (I don't know). I also side with using fabs since it is more explicit.

Reply:
I don't know which is more efficient, but I highly doubt the difference is measurable in any practical example.

Reply:
Just tested it, using a high-resolution c++ timer. With or without optimization, the latest version is slightly faster than the other two, but by no more than about 1%. So no worries...

Reply:
Nice! FWIF At first I thought both implementations would be "equivalent" and the compiler would be free to optimize them both into the same code. But this is not true, as `x^2 < y^2` does not imply `fabs(x) < fabs(y)` in floating point land:
```
x = 1e-200
y = 1e-300
x*x == y*y == 0.0
```